api/app/src/models/medical/facility.ts

```diff
@@ -0,0 +1,51 @@
+import { DataTypes, Sequelize } from "sequelize";
+import { OIDNode, OID_ID_START } from "../../shared/oid";
+import { BaseModel, defaultModelOptions, ModelSetup } from "../_default";
+import { Organization } from "./organization";
+
+export class Facility extends BaseModel<Facility> {
+  static NAME = "facility";
+  declare id: number;
+  declare organizationId: number;
+  declare systemRootOid: string;
+
+  static setup: ModelSetup = (sequelize: Sequelize) => {
+    Facility.init(
+      {
+        ...BaseModel.baseAttributes(),
+        id: {
+          type: DataTypes.INTEGER,
+          primaryKey: true,
+        },
+        organizationId: {
+          type: DataTypes.INTEGER,
+          primaryKey: true,
+        },
+        systemRootOid: {
+          type: DataTypes.STRING,
+          primaryKey: true,
+        },
+      },
+      {
+        ...defaultModelOptions(sequelize),
+        tableName: Facility.NAME,
+        hooks: {
+          async beforeCreate(attributes) {
+            const curMaxId = (await Facility.max("id", {
+              where: {
+                systemRootOid: attributes.systemRootOid,
+                organizationId: attributes.organizationId,
+              },
+            })) as number;
+            attributes.id = curMaxId ? (attributes.id = curMaxId + 1) : OID_ID_START;
+          },
+        },
```

Comment:
You decided to take this approach instead of DB trigger+procedure bc of Sequelize or because it was easier to build?

Just asking this because the performance implication: calculating the new ID in the DB is probably at least an order of magnitude lower than doing in the code - since this requires a request from the app to the DB.